issue with MIPS API

Trying to build the SNAPSHOT on bitrise failed, with a little more investigating with @bleege we were able to determine that this issue is specific to MIPS ABI, all others seem to work fine. Running `make apackage` locally resulted in this output:

```
...
CXX(target) Debug/obj.target/core/src/mbgl/geometry/static_vertex_buffer.o
  CXX(target) Debug/obj.target/core/src/mbgl/geometry/text_buffer.o
  CXX(target) Debug/obj.target/core/src/mbgl/geometry/vao.o
  CXX(target) Debug/obj.target/core/src/mbgl/gl/debugging.o
  CXX(target) Debug/obj.target/core/src/mbgl/gl/gl.o
  CXX(target) Debug/obj.target/core/src/mbgl/gl/gl_config.o
/var/folders/5r/8wwfkz3n4x11bcfpqjly7fjh0000gn/T/annotation_manager-84c2c5.s: Assembler messages:
/var/folders/5r/8wwfkz3n4x11bcfpqjly7fjh0000gn/T/annotation_manager-84c2c5.s:166400: Warning: no .cprestore pseudo-op used in PIC code
/var/folders/5r/8wwfkz3n4x11bcfpqjly7fjh0000gn/T/annotation_manager-84c2c5.s:166400: Warning: macro instruction expanded into multiple instructions
  CXX(target) Debug/obj.target/core/src/mbgl/gl/gl_object_store.o
/var/folders/5r/8wwfkz3n4x11bcfpqjly7fjh0000gn/T/annotation_manager-84c2c5.s:220949: Warning: no .cprestore pseudo-op used in PIC code
/var/folders/5r/8wwfkz3n4x11bcfpqjly7fjh0000gn/T/annotation_manager-84c2c5.s:220949: Warning: macro instruction expanded into multiple instructions
  CXX(target) Debug/obj.target/core/src/mbgl/gl/texture_pool.o
  CXX(target) Debug/obj.target/core/src/mbgl/layer/background_layer.o
clang36++: error: unable to execute command: Segmentation fault: 11
clang36++: error: assembler command failed due to signal (use -v to see invocation)
clang version 3.6 
Target: mipsel-none-linux-android
Thread model: posix
clang36++: note: diagnostic msg: PLEASE submit a bug report to http://source.android.com/source/report-bugs.html and include the crash backtrace, preprocessed source, and associated run script.
  CXX(target) Debug/obj.target/core/src/mbgl/layer/circle_layer.o
clang36++: note: diagnostic msg: 
********************

PLEASE ATTACH THE FOLLOWING FILES TO THE BUG REPORT:
Preprocessed source(s) and associated run script(s) are located at:
clang36++: note: diagnostic msg: /var/folders/5r/8wwfkz3n4x11bcfpqjly7fjh0000gn/T/annotation_manager-cd4044.cpp
clang36++: note: diagnostic msg: /var/folders/5r/8wwfkz3n4x11bcfpqjly7fjh0000gn/T/annotation_manager-cd4044.sh
clang36++: note: diagnostic msg: 

********************
make[1]: *** [Debug/obj.target/core/src/mbgl/annotation/annotation_manager.o] Error 254
make[1]: *** Waiting for unfinished jobs....
/var/folders/5r/8wwfkz3n4x11bcfpqjly7fjh0000gn/T/feature_index-bf683d.s: Assembler messages:
/var/folders/5r/8wwfkz3n4x11bcfpqjly7fjh0000gn/T/feature_index-bf683d.s:20314: Warning: no .cprestore pseudo-op used in PIC code
/var/folders/5r/8wwfkz3n4x11bcfpqjly7fjh0000gn/T/feature_index-bf683d.s:20314: Warning: macro instruction expanded into multiple instructions
/var/folders/5r/8wwfkz3n4x11bcfpqjly7fjh0000gn/T/feature_index-bf683d.s:23187: Warning: no .cprestore pseudo-op used in PIC code
/var/folders/5r/8wwfkz3n4x11bcfpqjly7fjh0000gn/T/feature_index-bf683d.s:23187: Warning: macro instruction expanded into multiple instructions
clang36++: error: unable to execute command: Segmentation fault: 11
clang36++: error: assembler command failed due to signal (use -v to see invocation)
clang version 3.6 
Target: mipsel-none-linux-android
Thread model: posix
clang36++: note: diagnostic msg: PLEASE submit a bug report to http://source.android.com/source/report-bugs.html and include the crash backtrace, preprocessed source, and associated run script.
clang36++: note: diagnostic msg: 
********************

PLEASE ATTACH THE FOLLOWING FILES TO THE BUG REPORT:
Preprocessed source(s) and associated run script(s) are located at:
clang36++: note: diagnostic msg: /var/folders/5r/8wwfkz3n4x11bcfpqjly7fjh0000gn/T/feature_index-7df367.cpp
clang36++: note: diagnostic msg: /var/folders/5r/8wwfkz3n4x11bcfpqjly7fjh0000gn/T/feature_index-7df367.sh
clang36++: note: diagnostic msg: 

********************
make[1]: *** [Debug/obj.target/core/src/mbgl/geometry/feature_index.o] Error 254
make: *** [android-lib-mips] Error 2
```

cc: @jfirebaugh @kkaefer @tmpsantos @brunoabinader @bleege

--- a/platform/android/bitrise.yml
+++ b/platform/android/bitrise.yml
@@ -31,6 +31,7 @@ workflows:
             #!/bin/bash
             set -eu -o pipefail
             apt-get install -y pkg-config
+            export BUILDTYPE=Release
             make android
             make test-android
     - slack:
@@ -71,6 +72,7 @@ workflows:
             signing.password=$SIGNING_PASSWORD
             signing.secretKeyRingFile=secring.gpg" >> platform/android/MapboxGLAndroidSDK/gradle.properties
 
+            export BUILDTYPE=Release
             make apackage -j4
 
             cd platform/android

